tests/components/automation/test_init.py

```diff
@@ -2878,6 +2882,68 @@ async def async_service_handler(service):
         assert "Disallowed recursion detected" not in caplog.text
 
 
+@pytest.mark.parametrize("wait_for_stop_scripts_after_shutdown", [True])
```

Comment:
I'm not sure this is a good thing, but we seem to explicitly want to allow infinite loops with restart automations are they are allowed to restart themselves